### PR TITLE
Make Mempool more configurable

### PR DESCRIPTION
**Pull request by malexandru-rdx**
Make mempool relayer and transaction reevaluation configurable and rename sizing fields to proper casing:

| Config Field | Docker Environment Variable | Default Value | Description |
| --- | --- | --- | --- |
| `mempool.relayer.interval_ms` | `RADIXDLT_MEMPOOL_RELAYER_INTERVAL_MS` | `20_000` | Interval time in ms between two relayer triggers |
| `mempool.relayer.max_peers` | `RADIXDLT_MEMPOOL_RELAYER_MAX_PEERS` | `100` | Maximum number of peers to relay to |
| `mempool.relayer.max_relayed_size` | `RADIXDLT_MEMPOOL_RELAYER_MAX_RELAYED_SIZE` | `6 * 1024 * 1024` | Maximum overall (to all peers) sent size on each trigger |
| `mempool.relayer.max_message_transaction_count` | `RADIXDLT_MEMPOOL_RELAYER_MAX_MESSAGE_TRANSACTION_COUNT` | `10` | Maximum number of transactions relayed to a peer |
| `mempool.relayer.max_message_payload_size` | `RADIXDLT_MEMPOOL_RELAYER_MAX_MESSAGE_PAYLOAD_SIZE` | `2 * 1024 * 1024` | Maximum payload size of a message relayed to a peer |
| `mempool.reevaluation.interval_ms` | `RADIXDLT_MEMPOOL_REEVALUATION_INTERVAL_MS` | `10_000` | Interval time in ms between two transaction reevaluation triggers |
| `mempool.reevaluation.max_count` | `RADIXDLT_MEMPOOL_REEVALUATION_MAX_COUNT` | `5` | Maximum number of transactions that will be *re-executed* every trigger  |
| `mempool.max_total_transactions_size` | `RADIXDLT_MEMPOOL_MAX_TOTAL_TRANSACTIONS_SIZE` |  `100 * 1024 * 1024` | Maximum total payload size of transactions living inside mempool |
| `mempool.max_transaction_count` | `RADIXDLT_MEMPOOL_MAX_TRANSACTION_COUNT` | `10_000` | Maximum number of transaction in the mempool |

### CODE REVIEW

File: core/src/main/java/com/radixdlt/mempool/MempoolRelayer.java [hunk 1]
```diff
@@ -132,7 +129,7 @@ public EventProcessor<MempoolRelayTrigger> mempoolRelayTriggerEventProcessor() {
     return ev -> {
       final var transactions =
           this.mempoolRelayReader.getTransactionsToRelay(
-              MAX_RELAY_MSG_NUM_TXNS, MAX_RELAY_MSG_TOTAL_TXN_PAYLOAD_SIZE);
+              (int) this.maxMessageTransactionCount, (int) this.maxMessagePayloadSize);
```
Comment (jakrawcz-rdx) [inline core/src/main/java/com/radixdlt/mempool/MempoolRelayer.java:132]: `Ints.checkedCast()` would make my paranoid side feel safer :p

alternatively (maybe even better), could it be parsed as `int` to begin with?

**Review by jakrawcz-rdx — APPROVED** [review]
Nice doc improvements too 👍 

I only have minor nitpicking comments.

File: core/src/main/java/com/radixdlt/p2p/transport/PeerChannelInitializer.java [hunk 2]
```diff
@@ -152,6 +125,7 @@ public final class PeerChannelInitializer extends ChannelInitializer<SocketChann
   private final Optional<RadixNodeUri> uri;
   private final Capabilities capabilities;
 
+  @Inject
```
[the remark this answers is not in the document]
Comment (malexandru-rdx) [reply core/src/main/java/com/radixdlt/p2p/transport/PeerChannelInitializer.java:128]: Ha, just a leftover, thanks. I tried to do it like that in the beginning but I just can't convince Juice to do what I'd like.

File: core/src/main/java/com/radixdlt/mempool/MempoolRelayerModule.java [hunk 3]
```diff
@@ -94,25 +94,31 @@ public void configure() {
...
 
   @ProvidesIntoSet
   private StartProcessorOnRunner mempoolRelayerStart(
-      EventProducer<MempoolRelayTrigger> dispatcher, @MempoolRelayMaxPeers int maxPeers) {
+      EventProducer<MempoolRelayTrigger> dispatcher,
+      @MempoolRelayMaxPeers int maxPeers,
+      @MempoolRelayerIntervalMs long mempoolRelayerIntervalMs,
+      @MempoolRelayerMaxMessageTransactionCount long mempoolRelayerMaxMessageTransactionCount,
+      @MempoolRelayerMaxMessagePayloadSize long mempoolRelayerMaxMessagePayloadSize,
+      @MempoolRelayerMaxRelayedSize long mempoolRelayerMaxRelayedSize) {
```
[the remark this answers is not in the document]
Comment (malexandru-rdx) [reply core/src/main/java/com/radixdlt/mempool/MempoolRelayerModule.java:109]: I'd much rather keep the verbose explanation for now.

File: core/src/main/java/com/radixdlt/mempool/MempoolRelayerConfig.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+package com.radixdlt.mempool;
+
+import com.google.inject.AbstractModule;
+
+/** Configuration parameters for mempool relayer. */
+public record MempoolRelayerConfig(
+    long intervalMs,
+    long maxRelayedSize,
+    long maxMessageTransactionCount,
+    long maxMessagePayloadSize) {
```
[the remark this answers is not in the document]
Comment (malexandru-rdx) [reply core/src/main/java/com/radixdlt/mempool/MempoolRelayerConfig.java]: Not really. Changed it :).

File: core/src/main/java/com/radixdlt/RadixNodeModule.java [hunk 5]
```diff
@@ -192,8 +192,33 @@ protected void configure() {
...
+                mempoolRelayerMaxRelayedSize,
+                mempoolRelayerMaxMessageTransactionCount,
+                mempoolRelayerMaxMessagePayloadSize)));
+
+    // Mempool Reevaluation
+    var mempoolReevaluationIntervalMs = properties.get("mempool.reevaluation.interval_ms", 10000);
+    var mempoolReevaluationMaxCount = properties.get("mempool.reevaluation.max_count", 5);
+    install(
+        new MempoolReevaluationModule(
+            Duration.ofMillis(mempoolReevaluationIntervalMs), mempoolReevaluationMaxCount));
```
[the remark this answers is not in the document]
Comment (malexandru-rdx) [reply core/src/main/java/com/radixdlt/RadixNodeModule.java:224]: I would do this in another refactor PR.

File: core/src/main/java/com/radixdlt/mempool/MempoolRelayer.java [hunk 6]
```diff
@@ -104,19 +92,28 @@ public final class MempoolRelayer {
       mempoolRelayReader;
 
   private final int maxPeers;
+  private final long maxMessagePayloadSize;
+  private final long maxMessageTransactionCount;
+  private final long maxRelayedSize;
```
Comment (jakrawcz-rdx) [inline core/src/main/java/com/radixdlt/mempool/MempoolRelayer.java:97]: (nitpick) these can all be just `int`s now I guess

**Review by jakrawcz-rdx — APPROVED** [review]
wow, you really went all the way with the config builders!

They look beautiful and verbose, as any Good Java code should 👍

**Pull request merged**